sdks/python/src/opik/evaluation/metrics/heuristics/contains.py

```diff
@@ -39,28 +59,37 @@ def __init__(
             track=track,
             project_name=project_name,
         )
-
         self._case_sensitive = case_sensitive
+        self._default_reference = reference
 
     def score(
-        self, output: str, reference: str, **ignored_kwargs: Any
+        self, output: str, reference: Optional[str] = None, **ignored_kwargs: Any
     ) -> score_result.ScoreResult:
         """
         Calculate the score based on whether the reference string is contained in the output string.
 
         Args:
             output: The output string to check.
-            reference: The reference string to look for in the output.
+            reference: The reference string to look for in the output. If None, falls back to the
+                default reference provided at initialization.
             **ignored_kwargs: Additional keyword arguments that are ignored.
 
         Returns:
             score_result.ScoreResult: A ScoreResult object with a value of 1.0 if the reference
                 is found in the output, 0.0 otherwise.
         """
+        # Use provided reference, else fall back to default
+        ref = reference if reference is not None else self._default_reference
```

Comment:
The bug is here, because you only check that reference is not None. Meantime it can be empty as well. You need to check for both: None and empty.